src/Handler/StreamHandler.php

```diff
@@ -286,7 +288,7 @@ private function createStream(RequestInterface $request, array $options)
             foreach ($options as $key => $value) {
                 $method = "add_{$key}";
                 if (isset($methods[$method])) {
-                    $this->{$method}($request, $context, $value, $params);
+                    $this->{$method}($request, $context, $value, $params, $uri);
```

Comment:
It is, but I chose to preserve the original URI in the request in case other callbacks, middleware etc. need it. The modified $url is not the actual URL being request but a version where the hostname has been replaced with an IP address, i.e. it lacks information about HTTP virtual host. This version is only used to "trick" fopen() into binding to the proper IP.